packages/language/examples/src/vesting.ts

```diff
@@ -0,0 +1,383 @@
+  ];
+
+  if (now > initialDepositDeadline && state?.accounts.length === 1)
+    return {
+      name: "NoDepositBeforeDeadline",
+      initialDepositDeadline: timeoutToDate(initialDepositDeadline),
+      payMinUtxoBackInput: [],
+    };
+
+  const environment = mkEnvironment(periodInterval[0])(periodInterval[1]);
+  const next = await getNext(environment);
+  if (next.can_reduce && emptyApplicables(next) && state?.accounts.length > 1)
+    return { name: "VestingEnded", withdrawInput: [] };
+
+  // Initial Deposit Phase
+  if (state?.accounts.length == 1) {
+    const depositInput =
+      next.applicable_inputs.deposits.length == 1
+        ? [Deposit.toInput(next.applicable_inputs.deposits[0])]
+        : undefined;
+    return {
+      name: "WaitingDepositByProvider",
+      initialDepositDeadline: timeoutToDate(initialDepositDeadline),
+      depositInput: depositInput,
+    };
+  }
+  if (
+    next.applicable_inputs.choices.length == 1 &&
+    next.applicable_inputs.choices[0].for_choice.choice_name == "cancel"
+  )
+    return {
+      name: "WithinVestingPeriod",
+      currentPeriod: currentPeriod,
+      periodInterval: periodInterval,
+      amountToWithdraw: 0n,
+      cancelInput: [Choice.toInput(next.applicable_inputs.choices[0])(1n)],
+    };
+  if (
+    next.applicable_inputs.choices.length == 1 &&
+    next.applicable_inputs.choices[0].for_choice.choice_name == "withdraw"
+  )
+    return {
+      name: "WithinVestingPeriod",
+      currentPeriod: currentPeriod,
+      periodInterval: periodInterval,
+      amountToWithdraw:
+        next.applicable_inputs.choices[0].can_choose_between[0].to,
+      withdrawInput: [
+        Choice.toInput(next.applicable_inputs.choices[0])(
+          next.applicable_inputs.choices[0].can_choose_between[0].to
+        ),
+      ],
+    };
+  if (next.applicable_inputs.choices.length == 2)
+    return {
+      name: "WithinVestingPeriod",
+      currentPeriod: currentPeriod,
+      periodInterval: periodInterval,
+      amountToWithdraw:
+        next.applicable_inputs.choices[0].can_choose_between[0].to,
+      cancelInput: [Choice.toInput(next.applicable_inputs.choices[1])(1n)],
+      withdrawInput: [
+        Choice.toInput(next.applicable_inputs.choices[0])(
+          next.applicable_inputs.choices[0].can_choose_between[0].to
+        ),
+      ],
+    };
+  return { unknownState: { request: request, state: state, next: next } };
+};
+
+const getPeriodInMilliseconds = function (frequency: Frequency): bigint {
+  switch (frequency) {
+    case "annually":
+      return 2n * getPeriodInMilliseconds("half-yearly");
+    case "half-yearly":
+      return 2n * getPeriodInMilliseconds("quarterly");
+    case "quarterly":
+      return 3n * getPeriodInMilliseconds("monthly");
+    case "monthly":
+      return 30n * getPeriodInMilliseconds("daily");
+    case "daily":
+      return 24n * getPeriodInMilliseconds("hourly");
+    case "hourly":
+      return 1n * 60n * 60n * 1000n;
```

Comment:
The numeric values do not exactly align with the English text. We should use calendar arithmetic instead.

Reply:
yeah, I don't like this... for unblocking the rest of the team, I would publish it wit this current implementation, but I want to use these kind of function provided by `date fns` : 
```js
// Each day between 6 October 2014 and 10 October 2014:
const result = eachDayOfInterval({
  start: new Date(2014, 9, 6),
  end: new Date(2014, 9, 10)
})
//=> [
//   Mon Oct 06 2014 00:00:00,
//   Tue Oct 07 2014 00:00:00,
//   Wed Oct 08 2014 00:00:00,
//   Thu Oct 09 2014 00:00:00,
//   Fri Oct 10 2014 00:00:00
// ]
``` 
They are aligning dates on the clock etc for hours, day, weekly  etc...